apps/files/src/components/FileEntry/FileEntryName.vue

```diff
@@ -187,76 +185,51 @@ export default defineComponent({
 				}
 			},
 		},
-	},
 
-	methods: {
-		/**
-		 * Check if the file name is valid and update the
-		 * input validity using browser's native validation.
-		 * @param event the keyup event
-		 */
-		checkInputValidity(event: KeyboardEvent) {
-			const input = event.target as HTMLInputElement
+		newName() {
+			// Check validity of the new name
 			const newName = this.newName.trim?.() || ''
-			logger.debug('Checking input validity', { newName })
-			try {
-				this.isFileNameValid(newName)
-				input.setCustomValidity('')
-				input.title = ''
-			} catch (e) {
-				if (e instanceof Error) {
-					input.setCustomValidity(e.message)
-					input.title = e.message
-				} else {
-					input.setCustomValidity(t('files', 'Invalid file name'))
-				}
-			} finally {
-				input.reportValidity()
+			const input = (this.$refs.renameInput as Vue|undefined)?.$el.querySelector('input')
+			if (!input) {
+				return
 			}
-		},
 
-		isFileNameValid(name: string) {
-			const trimmedName = name.trim()
-			const char = trimmedName.indexOf('/') !== -1
-				? '/'
-				: forbiddenCharacters.find((char) => trimmedName.includes(char))
-
-			if (trimmedName === '.' || trimmedName === '..') {
-				throw new Error(t('files', '"{name}" is an invalid file name.', { name }))
-			} else if (trimmedName.length === 0) {
-				throw new Error(t('files', 'File name cannot be empty.'))
-			} else if (char) {
-				throw new Error(t('files', '"{char}" is not allowed inside a file name.', { char }))
-			} else if (trimmedName.match(window.OC.config.blacklist_files_regex)) {
-				throw new Error(t('files', '"{name}" is not an allowed filetype.', { name }))
-			} else if (this.checkIfNodeExists(name)) {
-				throw new Error(t('files', '{newName} already exists.', { newName: name }))
+			let validity = getFilenameValidity(newName)
```

Comment:
One tiny change I encountered last week: always reset the `setCustomValidity` before setting it again.
Otherwise if some conditions changes, the state stays the same.

```suggestion
			// Reset custom validity
			input.setCustomValidity('')

			let validity = getFilenameValidity(newName)
```

Reply:
It is already reset as the function will be `''` if valid -> reset.

Reply:
Ah, a bit hidden! 
And as isRenaming will destroy the input, we don't need to clear it otherwise :+1: 

Good from me!